:alembic: Migrade Demo SDK to KMP

krllus opened the pull request:
# Descrição

<!-- Por favor, forneça abaixo uma breve descrição das mudanças incluídas neste Pull Request -->



---

# Checklist

<!-- Verifique os itens abaixo e preencha com um X entre os colchetes nos que foram concluídos antes de enviar -->

- [ ] Testei as alterações localmente.
- [ ] Atualizei qualquer documentação relevante.
- [ ] Verifiquei que não há problemas de Lint ou formatação.
- [ ] Confirmei que o código segue as diretrizes de contribuição do projeto.

krllus committed:
[PAIPA#10968] Send Receipt by email (#71)

* adding send mail into list;

* bump libraries version because new SDK dependency;

* changing cancel map from itk to atk

* adjusts for email provider

* adding logs;

* adjusting email provider;

* adjusting email provider;

* add new sdk version

---------

Co-authored-by: italo.santana <italo.santana@stone.com.br>

krllus committed:
[PAIPA#11443] Ajuste de envio de email para cliente e lojista (#72)

krllus committed:
Merge branch 'migrate_demo_to_kmp' into sdk_v6/change_sdk_version

krllus committed:
adjusting packaging information

krllus committed:
adjusting imports

krllus committed:
adjusting application context being required and moving SDK initialization into DI Framework initialization;

krllus committed:
attempt to fix debug duplication error when compiling code;

krllus committed:
removing not used imports

krllus committed:
adjustments in catalog versions

krllus committed:
adjustments for duplicated classes when building

krllus committed:
adjusting DI for wrappers and view models

krllus committed:
Adjust DI for BluetoothDeviceRepository in view models and wrappers.

krllus committed:
fix material icons not being found for base spinner

krllus committed:
Update dependencies and adjust versioning for lifecycle and navigation components

krllus committed:
Remove unnecessary else block in ManageStoneCodeLayout

krllus committed:
improving common modules dependencies;

krllus committed:
Remove unused imports from KoinModule.apple.kt

krllus committed:
Remove unused import;

krllus committed:
fix package name

krllus committed:
fix bluetooth datastore preference;

marcostark commented on app/src/commonMain/kotlin/br/com/stonesdk/sdkdemo/ui/splashscreen/ValidationLayout.kt:
```diff
@@ -71,7 +64,7 @@ internal fun ValidationScreen(
                     Text("Error: $code - $message")
                     Button(onClick = {
-                        viewModel.initializeSDK(context, appInfo)
+                        //viewModel.initializeSDK(context, appInfo)
```
pode ser removido esse comentario ?

krllus replied:
acho que pode ser removido todo o componente do botao;
essa acao de inicializacao do SDK foi removida do viewmodel.

marcostark commented on app/src/commonMain/kotlin/br/com/stonesdk/sdkdemo/ui/transactions/TransactionViewModel.kt:
```diff
@@ -82,46 +85,76 @@ class TransactionViewModel(
+
+            val connectedDevice = bluetoothRepository.getConnectedBluetoothDevice()
+            if (connectedDevice == null) {
+                // TODO display error message indicating bluetooth is not connected
```
esse TODO já foi implementado ?

krllus replied:
ainda nao.

krllus committed:
🚧 Ajustes navegacao Demo SDK (#74)

* adjusting code because of update in SDK and HAL internal implementation;

* adjusting theme surface for dark mode;

* layout adjustments;

* layout adjustments;

* fix compilation error for email provider.

* change gradle configuration; trying to improve build time;

* adjusting navigation and components location

* removing log imports as they are not being used;

* fix paste and type of non digits in activation;

* adjustments in navigation between screens

* removing gertec from tasks to exclude when assemble

italomarcel committed:
Merge branch 'migrate_demo_to_kmp' into sdk_v6/change_sdk_version

italomarcel committed:
Update Kotlin and desugar versions in libs.versions.toml

italomarcel committed:
fix iOS transaction button

italomarcel committed:
Update SDK version to 0.1.10 and improve transaction layout

italomarcel committed:
Update SDK version to 0.1.11 and improve error handling in transaction processing

krllus committed:
Update SDK version to 6.0.0 and remove beta and alpha suffixes

marcostark committed:
Merge pull request #76 from stone-payments/bug/PAIPA-12983-send-email

🐛 [PAIPA-12983] Fix sending email

marcostark committed:
chore: add build-all.sh for parallel Android+iOS debug builds

Usage:
  ./build-all.sh            # both in parallel (default)
  ./build-all.sh android    # Android only
  ./build-all.sh ios        # iOS only (simulator)

Logs go to build/logs/{android,ios}-debug.log; iOS derived data goes to
build/ios-derived-data so it does not pollute the global DerivedData.

marcostark committed:
chore: update SDK version to 6.0.3-RC19

marcostark committed:
chore: change environment from CERTIFICATION to STAGING in DemoApplication and KoinModule
